src/main/java/org/truffleruby/core/numeric/IntegerNodes.java

```diff
@@ -1160,6 +1160,15 @@ protected Object leftShiftNeg(int a, long b) {
             return rightShiftNode.executeRightShift(a, absoluteValue(b));
         }
 
+        @Specialization(guards = "b < 0")
+        protected Object leftShiftNeg(long a, long b) {
+            if (rightShiftNode == null) {
+                CompilerDirectives.transferToInterpreterAndInvalidate();
+                rightShiftNode = insert(RightShiftNodeFactory.create(null));
+            }
+            return rightShiftNode.executeRightShift(a, absoluteValue(b));
+        }
```

Comment:
Does << need a similar fix?
A spec is probably best here to ensure we don't regress.